animating-widgets.md

```diff
@@ -0,0 +1,293 @@
+between going forward or reverse.
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:flutter/animation.dart';
+
+class LogoApp extends AnimatedWidget {
+
+  // Declare the Tween objects as static because we don't change them.
+  static final _opacityTween = new Tween<double>(begin: 1.0, end: 0.3);
+  static final _sizeTween = new Tween<double>(begin: 0.0, end: 300.0);
+
+  LogoApp({Key key, Animation<double> animation})
+      : super(key: key, animation: animation);
+
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Opacity(
+        opacity: _opacityTween.evaluate(animation),
+        child: new Container(
+          margin: new EdgeInsets.symmetric(vertical: 10.0),
+          height: _sizeTween.evaluate(animation),
+          width: _sizeTween.evaluate(animation),
+          child: new AssetImage(
+            name: 'assets/flutter_logo.png',
+            fit: ImageFit.contain
+          )
+        )
+      )
+    );
+  }
+}
+
+void main() {
+  final AnimationController controller = new AnimationController(
+    duration: const Duration(milliseconds: 5000)
+  );
+
+  final Animation<double> animation =
+      new CurvedAnimation(parent: controller, curve: Curves.bounceIn);
+
+  animation.addStatusListener((status) {
+    if (status == AnimationStatus.completed)
+      controller.reverse();
+    else if (status == AnimationStatus.dismissed)
+      controller.forward();
+  });
+
+  runApp(
+    new MaterialApp(
+      title: 'My Animation Test',
+      routes: <String, WidgetBuilder> {
+        '/': (BuildContext context) => new LogoApp(
+          animation: animation
+        )
+      }
+    )
+  );
+  controller.forward();
+}
+```
+
+## Dispose your widget animation
+
+It’s good practice to dispose your animation controller and free up your
+resources when they are no longer needed. To do so, call the
+[`dispose()`](http://docs.flutter.io/flutter/material/AnimationController/dispose.html)
+method on the
+[`AnimationController`](http://docs.flutter.io/flutter/material/AnimationController-class.html)
+object for the animation object you want to free. This stops the animation from
+running.
+
+The following snippet shows how to call the animation controller’s
+[`dispose()`](http://docs.flutter.io/flutter/material/AnimationController/dispose.html)
+method from within a
+[`State`](http://docs.flutter.io/flutter/material/State-class.html) object’s
+[`dispose()`](http://docs.flutter.io/flutter/material/State/dispose.html) method
+implementation. The framework typically invokes the
+[`State`](http://docs.flutter.io/flutter/material/State-class.html)
+object’s [`dispose()`](http://docs.flutter.io/flutter/material/State/dispose.html)
+method when switching between screens, as directed by the routes property.
+
+```dart
+class LogoAppState extends State<LogoApp> {
+
+  // Implementation code omitted for brevity.
+
+  @override
+  void dispose() {
+     controller.dispose();
+     super.dispose();
+  }
```

Comment:
We should be doing this for all the examples throughout.  It's not really optional.